BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Endpoints/EndpointProvider.cs

```diff
@@ -129,46 +128,9 @@ public string GetInstrumentationKey()
         /// </summary>
         /// <remarks>Example: "key1=value1;key2=value2;key3=value3".</remarks>
         /// <returns>A dictionary parsed from the input connection string.</returns>
-        internal static Dictionary<string, string> ParseConnectionString(string connectionString)
+        internal static IDictionary<string, string> ParseConnectionString(string connectionString)
         {
-            if (connectionString == null)
-            {
-                CoreEventSource.Log.ConnectionStringNull();
-                throw new ArgumentNullException(nameof(connectionString));
-            }
-
-            var keyValuePairs = connectionString.Split(SplitSemicolon, StringSplitOptions.RemoveEmptyEntries);
-
-            if (keyValuePairs.Length == 0)
-            {
-                CoreEventSource.Log.ConnectionStringEmpty();
-                throw new ArgumentException("Connection string cannot be empty.");
-            }
-
-            var dictionary = new Dictionary<string, string>(keyValuePairs.Length, StringComparer.OrdinalIgnoreCase);
-
-            foreach (var pair in keyValuePairs)
-            {
-                var keyAndValue = pair.Split('=');
-                if (keyAndValue.Length != 2)
-                {
-                    CoreEventSource.Log.ConnectionStringInvalidDelimiters();
-                    throw new ArgumentException("Connection String Invalid: Unexpected delimiter can not be parsed. Expected: 'key1=value1;key2=value2;key3=value3'");
-                }
-
-                var key = keyAndValue[0].Trim();
-                var value = keyAndValue[1].Trim();
-
-                if (dictionary.ContainsKey(key))
-                {
-                    CoreEventSource.Log.ConnectionStringDuplicateKey();
-                    throw new ArgumentException(FormattableString.Invariant($"Connection String Invalid: Contains duplicate key: '{key}'."));
-                }
-
-                dictionary.Add(key, value);
-            }
-
-            return dictionary;
+            return ConfigStringParser.Parse(connectionString, configName: "Connection String");
```

Comment:
catch and log Error here - ConnectionString parse failure is pretty bad and we want it Error level.

Reply:
done